File: hospital/model/dao/MedicalRecordDAO.js
/**
 * DAO Interface: MedicalRecordDAO
 */
export default class MedicalRecordDAO {
  async getAll()                    { throw new Error('MedicalRecordDAO.getAll() not implemented'); }
  async getById(id)                 { throw new Error('MedicalRecordDAO.getById() not implemented'); }
  async getByPatient(patientId)     { throw new Error('MedicalRecordDAO.getByPatient() not implemented'); }
  async getByDoctor(doctorId)       { throw new Error('MedicalRecordDAO.getByDoctor() not implemented'); }
  async getByAppointment(apptId)    { throw new Error('MedicalRecordDAO.getByAppointment() not implemented'); }
  async create(record)              { throw new Error('MedicalRecordDAO.create() not implemented'); }
  async update(id, data)            { throw new Error('MedicalRecordDAO.update() not implemented'); }
  async delete(id)                  { throw new Error('MedicalRecordDAO.delete() not implemented'); }
}

File: hospital-backend/src/main/java/com/hospital/dao/MedicalRecordDAO.java
package com.hospital.dao;

import com.hospital.model.MedicalRecord;
import java.util.List;

public interface MedicalRecordDAO {
    List<MedicalRecord> getAll();
    MedicalRecord getById(int id);
    List<MedicalRecord> getByPatientId(int patientId);
    List<MedicalRecord> getByDoctorId(int doctorId);
    int insert(MedicalRecord record);
    int update(MedicalRecord record);
    int delete(int id);
}
